presto-main/src/main/java/io/prestosql/operator/TableScanOperator.java

```diff
@@ -293,6 +293,9 @@ public Page getOutput()
             return null;
         }
         if (source == null) {
+            if (!dynamicFilter.get().isAll()) {
```

Comment:
there is also `TableScanWorkProcessorOperator`

Reply:
Made changes there as well